ofono/include/modem.h

```diff
@@ -142,6 +142,9 @@ ofono_bool_t ofono_modem_remove_watch(unsigned int id);
 void ofono_modem_set_powered_timeout_hint(struct ofono_modem *modem,
 							unsigned int seconds);
 
+/* Since mer/1.28+git4 */
+unsigned int ofono_modem_get_next_call_id(struct ofono_modem *modem);
```

Comment:
Simply: to get the id of the current call in order to be able to register for signals from the proper voicecall interface. I know that there is another way, just to listen when the interface is up as call is initiated but it did seem to have timing issues if the service that is being started relies on that info.

It turned out that the signal "active" from the voicecall interface could be missed in some cases because of timing. I found this to be the reliable solution here, to ensure that the service started by the plugin knows exactly the correct interface to listen for call state signals. It was used by existing code as well so this is just a getter for the same, in order to be able to tell the service started by the plugin the exact info.

Reply:
> Why would you need to register signal handlers for the next call? Which may never occur. And for the changes in the current call list, you get a D-Bus signal with the actual id of the current call.

Because in the situation when the call is being initiated and the plugin that is used to filter the call in some way, like in AML, does get the call info before it is being added. So the next call id is the call id for this particular call that is being made, I observed and debugged this to be a fact. Then afterwards ofono increases the counter but only after the plugin has returned doing the filtering for the call. AFAIK there cannot be two calls made in the same time and both waiting, another one is on hold so in this case the naming might be misleading as the current call id = next one from the list, which will be properly assigned.

This was due to getting the interface info at startup and the timing issue was from call to D-Bus -> start service -> listen for changes ->> delay that call may have been already answered and is ongoing already - in some countries AML sequence starts from 0, so they expect the message right after answering the call. With this call id passing this is not an problem.

I can try to implement something like that with synchronous calls as it is kind of moot having to wait anything at that stage regarding the AML service sole purpose being to be initiated when the call is established. I already tried quite similar approach but it was I month ago and this did work like charm so I did not bother with that anymore.

If the libgofono wouldn' require too much changes to the code it would be good to have some way to query the call state but how otherwise the call id can be retrieved? We have to have some kind of identification for the call to request it. There can be calls on hold on the same modem, right?

Reply:
I'm thinking along these lines:

The plugin uses the "emergency" status of the call as a criteria for starting the AML service. Call id is not involved.

Once the service gets started, is has to get the list of the current calls, find the emergency call and start tracking it. Or exit if the emergency call has terminated by then. Again, call ids are not involved - calls are identified by object paths.

Reply:
Well `CallAdded` comes too early. It is not possible for the service to catch it. The call id would have been to listen to the correct voicecall interface for the messages and it was really straightforward to pass that as an argument. Instead of "service A starts service B over D-Bus and requires service B to request from service A for more info" ... I dislike this, if it is a matter of few parameters. But anyways, dropped this and made changes to the other service "B".